Compiler AssertionError when redefining java.lang.String

I get an AssertionError during compilation when redefining java.lang.String as follows:

```scala
    package java
    package lang
    object String
```

```scala
[info] Compiling 1 Scala source to /home/olivier/workspace/reflect/out/target/scala-2.10/classes...
[error] uncaught exception during compilation: java.lang.AssertionError
[trace] Stack trace suppressed: run last compile:compile for the full output.
[error] (compile:compile) java.lang.AssertionError: assertion failed: 
[error]      while compiling: /home/olivier/workspace/reflect/out/java.lang.String.scala
[error]         during phase: jvm
[error]      library version: version 2.10.2
[error]     compiler version: version 2.10.2
[error]   reconstructed args: -bootclasspath /usr/lib/jvm/java-7-oracle/jre/lib/resources.jar:/usr/lib/jvm/java-7-oracle/jre/lib/rt.jar:/usr/lib/jvm/java-7-oracle/jre/lib/sunrsasign.jar:/usr/lib/jvm/java-7-oracle/jre/lib/jsse.jar:/usr/lib/jvm/java-7-oracle/jre/lib/jce.jar:/usr/lib/jvm/java-7-oracle/jre/lib/charsets.jar:/usr/lib/jvm/java-7-oracle/jre/lib/jfr.jar:/usr/lib/jvm/java-7-oracle/jre/classes:/home/olivier/.sbt/boot/scala-2.10.2/lib/scala-library.jar -classpath /home/olivier/workspace/reflect/out/target/scala-2.10/classes
[error] 
[error]   last tree to typer: Select(This(package lang), String)
[error]               symbol: object String in package lang (flags: <module> <java> <triedcooking>)
[error]    symbol definition: object String
[error]                  tpe: String.type
[error]        symbol owners: object String -> package lang
[error]       context owners: object String -> package lang
[error] 
[error] == Enclosing template or block ==
[error] 
[error] Template( // val <local String>: <notype> in object String, tree.tpe=String.type
[error]   "java.lang.Object", "scala.Serializable" // parents
[error]   ValDef(
[error]     private
[error]     "_"
[error]     <tpt>
[error]     <empty>
[error]   )
[error]   // 2 statements
[error]   DefDef( // private def readResolve(): Object in object String
[error]     <method> private <synthetic>
[error]     "readResolve"
[error]     []
[error]     List(Nil)
[error]     <tpt> // tree.tpe=Object
[error]     lang.this."String" // object String in package lang, tree.tpe=String.type
[error]   )
[error]   DefDef( // def <init>(): String.type in object String
[error]     <method> <java>
[error]     "<init>"
[error]     []
[error]     List(Nil)
[error]     <tpt> // tree.tpe=String.type
[error]     Block( // tree.tpe=Unit
[error]       Apply( // def <init>(): Object in class Object, tree.tpe=Object
[error]         String.super."<init>" // def <init>(): Object in class Object, tree.tpe=()Object
[error]         Nil
[error]       )
[error]       ()
[error]     )
[error]   )
[error] )
[error] 
[error] == Expanded type of tree ==
[error] 
[error] TypeRef(TypeSymbol(class String extends Serializable))
[error] 
[error] how can getCommonSuperclass() do its job if different class symbols get the same bytecode-level internal name: java/lang/String
```

--- a/test/files/pos/t8093.scala
+++ b/test/files/pos/t8093.scala
@@ -1,0 +1,4 @@
+package java
+package lang
+
+object String

